tests/Tutor.Infrastructure.Tests/Unit/EventStore/QueryInterfaceTests.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Shouldly;
+using Tutor.Infrastructure.Database.EventStore;
+using Tutor.Infrastructure.Tests.TestData.EventStore;
+using Xunit;
+
+namespace Tutor.Infrastructure.Tests.Unit.EventStore
+{
+    public class QueryInterfaceTests : IClassFixture<EventStoreManager>
+    {
+        private readonly IEventStore _eventStore;
+
+        public QueryInterfaceTests(EventStoreManager eventStoreManager)
+        {
+            _eventStore = eventStoreManager.EventStore;
+        }
+
+        [Theory]
+        [MemberData(nameof(QueryTestData.GetQueries), MemberType = typeof(QueryTestData))]
+        public void After_before_conditions(Query parameters)
+        {
+            var _expectedResult = parameters.GetExpectedResult(QueryTestData._allEvents);
+
+            var query = _eventStore.Events;
```

Comment:
Do we need the event store for these tests, especially since we are targeting unit tests?

Maybe just test the PostgresEventQueryable directly?

As a side note, is `PostgresEventQueryable` the best name? I'm not sure if it has any postgres-specific code?

Reply:
We don't actually need any other event store functionalities in this test class, I changed it to just have an IEventQueryable field instead in 125b47c6a721df6ecd9e083acde1a46321a70a8f. 
PostgresEventQueryable is Postgres specific since it used the Json field, I don't think any of the code in the PostgresStore package would work with another EFCore driver because of that. 
These tests aren't specific for PostgresEventQueryable though, they would be the same for any implementation of IEventQueryable. I initially wanted to inject the IEventQueryable through test data, but I didn't find a way to do that with proper database seeding and teardown.